Arnom ms german translations part1 (#218)
InnerSourceCommons/InnerSourceLearningPath: introduction/de/03-how-works.asciidoc

== Wie funktioniert InnerSource?

Angenommen, Team A verwendet Software, die von Team B produziert wird.
Team A sendet eine Funktionsanforderung an Team B, aber Team B kann diese Funktion nicht rechtzeitig für Team A implementieren.

> In einer InnerSource Umgebung kann Team A anstatt einer Feature-Anforderung auch alternativ einen PullRequest an Team B zur Integration schicken.

Reviewer: ```suggestion
Nehmen wir folgendes Szenario an: Team A verwendet Software, die von Team B produziert wird.
Team A übermittelt Anforderungen an Team B, aber Team B ist stark ausgelastet und kann diese nicht rechtzeitig implementieren.
InnerSource ermögicht Team A, anstatt einer Anforderungen, einen PullRequest mit der eigenen Implementierung an Team B zu schicken.
```

Reviewer: I havent been involved with the Learning Path lately. Was it conscious decision (in the English version) to use `Pull Request` (Github specific term) vs. something more general like `code contribution`, `change proposal`, ....

Reviewer 2: I feel like this term is universally-understood and the clearest way to express what we're talking about.  If I saw one of those other terms I feel like people would want to ask something like `You mean a pull request?`.  _BitBucket_ and _GitLab_ both mention pull requests as well.

Reviewer 2: I'm willing to be proved wrong, though, if there's reason to believe that the term "pull request" might be confusing to some.

Author: All git systems I know use pull request as the term. "git system" may be the limitation here, but I think today it is safe to say they are vastly dominating for version control, no?  Pull request is really the mechanics, the term "sending a pull request" however I think at this point can be used exchangeable with sending a code contribution. (just like paper handkerchief and Kleenex)..

Das heißt, Team A implementiert die Funktion direkt in der Software von Team B und sendet einen PullRequest mit den Codeänderungen.
In enger Partnerschaft überprüft Team B dann den übermittelten Code und ggf. akzeptiert die Aenderungen wenn sie den Anforderungen enpricht.

> In diesem Beispiel nennen wir Team A das _Guest_-Team und Team B das _Host_-Team.

Reviewer 3: I argue that having some common foreign language terms is a good way. Going all Paketvermittlungsknoten, Gruppenruf, Funktionseinschub und Fegelinienalgorithmus never really helped anyone over a longer period of time.

> In diesem Beispiel nennen wir Team A das _Guest_-Team und Team B das _Host_-Team.

Author: So I guess we leave it as is?  (resolving in this spirit

* Der Produktbesitzer stellt sicher, dass die Featureanforderungen gemaess des Entwicklungsprozzesses des Hostteams  erstellt werden, entweder von Mitgliedern des Gastteams oder Mitgliedern des Gastgeberteams. 
Diese Documentation geschieht meist in der Form von Userstories, welche die gewünschte Funktion und Rahmenbedingungen beschreiben, die das Gastteam akzeptiert hat.
Sie listet auch alle Details aus Sicht des Host-Teams auf, z.B. wie eine Funktion geschrieben und eingereicht werden soll damit der Code akzeptiert werden kann.
Beispiele für solche Details sind Architektureinschränkungen, Codierungskonventionen, Abhängigkeitsverwendungen, Datenverträge usw.
* Unterstützt vom vertrauenswürdigen Committer, sendet der Mitwirkende den PullRequest, um die angefragte Funktion zu implementieren.

Beachten Sie, dass bei diesen Schritten kein bestimmtes System für die allgemeine Organisation der Zeit oder der Prioritäten eines Teams angenommen wird. 
InnerSource geht davon aus, dass Teams bereits über vorhandene Organisationsmethoden verfügen, und bietet einen Rahmen für die Zusammenarbeit, wenn ein Gastteam Code zu einem Project beitragen möchte.

> Dieser Weg der Zusammenarbeit funktioniert gut für das Gastteam, da es die Funktionalität erhält die es benötigt, wenn sie sie benötigt, und ohne die Notwendigkeit die langfristige Wartung der Lösung zu übernehmen.

Reviewer: ```suggestion
Dieser Weg der Zusammenarbeit birgt Vorteile für das Gastteam. Es erhält die Funktionalität, die es benötigt zum rechten Zeitpunkt und ohne die Verpflichtung, die langfristige Wartung der Lösung zu übernehmen.
```

> Dieser Weg funktioniert für das Gastgeberteam, weil es in der Lage ist, besser zu skalieren und seine Kunden besser zu bedienen.

Reviewer: `Kunden` feels not ideal here. I can't find a better term, though.

Author: In another round I think there is room for some more wordsmithing.

Dies funktioniert für das Unternehmen in der Gesamtheit, weil Lösungen für gemeinsame Probleme an einem gemeinsamen, zentral gepflegten Ort jedem zur Verfuegung gestellt werden.

> Letzlich steht mehr Zeit für die Erstellung von Code zur Verfügung, der die wirklichen Unternehmensprobleme löst, anstatt sie zur Verhandlung über Features oder für Eskalationsprozesse zu verwenden.

Reviewer: ```suggestion
In letzter Konsequenz fließt mehr Zeit in Code, der die eigentlichen Unternehmensprobleme löst, anstatt in Verhandlungen über Features oder in Eskalationsprozesse.
```
